### DISCUSSION TITLE
I am the author of the Vlpp library but I think the CMakeLists.txt does not make sense in the package.

### DISCUSSION THREAD

**Discussion by vczh** [microsoft/vcpkg#26825] [Q&A]
So I just know my library is in vcpkg 4 years after it was submitted here. The problem is that, the library cannot be simply compiled as a library as what `CMakeLists.txt` described, because

- Files should be selected based on what OS you are working on.
- Files should be selected based on the preprocessor you add to your project, for example, if you add `VCZH_DEBUG_NO_REFLECTION`, then some files must be eliminated. And such preprocessor should be apply to both Vlpp and the application using it.
- If you don't need all the feature, you don't have to include all qualified files.
- It would be at least 36 different combinations of configurations.

My designed way to use this library is that, people should create lib projects based on what they need by themselves, or just add files they need to their projects directly. Does vcpkg support such scenario, that I have multiple projects using Vlpp, but each project has a different configuration, so different files and preprocessors are used when compiling the package? And how to achieve that?

Sorry because I have never used vcpkg and cmake before.

**Comment by JackBoosY** [comment]
So my PR fixed this issue right?

**Comment by vczh** [comment]
Yes
